File: API/Features/Ledgers/ViewModels/LedgerVM.cs
using System.Collections.Generic;
using API.Infrastructure.Classes;

namespace API.Features.Ledger {

    public class LedgerVM {

        public SimpleEntity Customer { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPersons { get; set; }
        public int TotalEmbarked { get; set; }
        public IEnumerable<LedgerPortVM> Ports { get; set; }
        public List<LedgerReservationVM> Reservations { get; set; }

    }

}

File: FrontEnd/src/app/features/ledgers/classes/view-models/list/ledger-vm.ts
import { CustomerVM } from './ledger-customer-vm'
import { LedgerPortGroupVM } from './ledger-port-group-vm'
import { LedgerReservationVM } from './ledger-reservation-vm'

export interface LedgerVM {

    customer: CustomerVM
    ports: LedgerPortGroupVM[]
    reservations: LedgerReservationVM[]

}
